### PR TITLE
Don't stop scanning album on media fail

### PR DESCRIPTION
**Pull request by kkovaletp**
Reimplementing PR #820, as the author removed his repo, but I think that this change is important:

Stopping scan after a media failed is also skipping all following files. The album will only show content up to the failed item with no simple option to recover from this situation.

Instead when continuing the scan the whole album will be scanned and failed items will show up with a missing thumbnail. That way at least everything that can be scanned is scanned, and failed media items might still show fine in the original (e.g. Samsung panorama photos).

### CODE REVIEW

**Comment by emersion** [comment]
LGTM

**Comment by sigmonsays** [comment]
is this ever gonna be merged?

is development stopped on this project?

**Comment by kkovaletp** [comment]
> is development stopped on this project?

yes, check this discord channel for details https://discord.com/channels/794576839813234688/794576839813234696

**Comment by DaCHack** [comment]
> > is development stopped on this project?
> 
> yes, check this discord channel for details https://discord.com/channels/794576839813234688/794576839813234696

Just created an account but get the message that I do Not have access to text channels or the server has none. Is it me or is the channel offline, @kkovaletp ?

**Comment by kkovaletp** [comment]
> Is it me or is the channel offline

That is something on your side: I see channels and the content inside

**Comment by DaCHack** [comment]
> > Is it me or is the channel offline
> 
> That is something on your side: I see channels and the content inside

Is it maybe invite-only? Could you send an invite link?

**Comment by kkovaletp** [comment]
> invite link

https://discord.gg/s35EXVT8

**Comment by DaCHack** [comment]
> > invite link
> 
> https://discord.gg/s35EXVT8

Works now. Many thanks

**Comment by kkovaletp** [comment]
I've merged this PR into #910 - a temporary alternative to the master while the project is on hold, so users still could get the latest Photoview improvements and fixes by building an image locally from this PR code

**Review by viktorstrate — APPROVED** [review]
This should definitely be merged to fix it, but ideally an error notification should be send to the user and not just be logged, but think that should be a future improvement.

**Comment by kkovaletp** [comment]
@jordy2254 , what do you think about the comment from @viktorstrate 

> ideally an error notification should be send to the user and not just be logged

Do you see an easy way to add the user notification from a running job to the UI in a similar way as the job progress is shown (but with some warning sign, so a user can visually differentiate the message for regular status updates)?

I think that if there is a simple and straightforward way of doing that, we could include it into the PR before merging it. If not, I'll create an issue to not forget about this improvement in the future

**Comment by jordy2254** [comment]
Implementation of the notification should be really easy. the findMediaForAlbum already implements this for the initilise of media to scan and process. We should do something similar and improve the message to better reflect what album and file was the problem.
https://github.com/photoview/photoview/pull/892/files#diff-46ce03b017c3c7c7fd35cc530af92c87f97c818314b1182d0a02e1e3c8fcc1d2R160

**Comment by kkovaletp** [comment]
@jordy2254 
Great! If you can do a code review and propose the change for this PR, adding this code, it would be great, so I'll accept it, test it locally, and then merge it all together as a completed Product Increment

File: api/scanner/scanner_album.go
Comment (jordy2254) [inline api/scanner/scanner_album.go:106]:
```suggestion
	scanner_utils.ScannerError("Error scanning media for album (%d) file (%s): %s\n", ctx.GetAlbum().ID, media.Path, err)		
```

**Comment by kkovaletp** [comment]
I've accepted your change request, @jordy2254, but also left the logging as well with the enhanced error message - I think that it should be logged anyway.

**Comment by kkovaletp** [comment]
I did local testing of the new change and it works fine: an error is shown in the UI and is logged.
The error message in the UI looks a bit ugly, as the text goes out of the pop-up border, but this is a minor issue, which can be fixed by another PR later:
Снимок экрана 30 03 2024 в 11 25 56

**Comment by jordy2254** [comment]
I shall raise an issue to address some improvements to the notifications. We don't seem to do anything with an error state based on that screenshot. Looks good now though thanks.

**Review by jordy2254 — APPROVED** [review]

**Pull request merged**
